src/Akka.Persistence.Sql.Hosting.Tests/CustomSqlEndToEndSpec.cs

```diff
@@ -0,0 +1,275 @@
+
+            var customSnapshot2 = await customMyPersistentActor2.Ask<int[]>(GetAll, timeout);
+            customSnapshot2.Should().BeEquivalentTo(new[] { 1, 2 });
+            
+            // validate configs
+            var config = Sys.Settings.Config;
+            config.GetString("akka.persistence.journal.plugin").Should().Be("akka.persistence.journal.sql");
+            config.GetString("akka.persistence.snapshot-store.plugin").Should().Be("akka.persistence.snapshot-store.sql");
+
+            var customJournalConfig = config.GetConfig("akka.persistence.journal.custom");
+            customJournalConfig.Should().NotBeNull();
+            customJournalConfig.GetString("connection-string").Should().Be(_fixture.ConnectionString);
+            customJournalConfig.GetString("provider-name").Should().Be(_fixture.ProviderName);
+
+            var customSnapshotConfig = config.GetConfig("akka.persistence.snapshot-store.custom");
+            customSnapshotConfig.Should().NotBeNull();
+            customSnapshotConfig.GetString("connection-string").Should().Be(_fixture.ConnectionString);
+            customSnapshotConfig.GetString("provider-name").Should().Be(_fixture.ProviderName);
+
+            // validate that query is working
+            var readJournal = Sys.ReadJournalFor<SqlReadJournal>("akka.persistence.query.journal.sql");
+            var source = readJournal.AllEvents(Offset.NoOffset());
+            var probe = source.RunWith(this.SinkProbe<EventEnvelope>(), Sys.Materializer());
+            probe.Request(2);
+            probe.ExpectNext<EventEnvelope>(p => p.PersistenceId == PId && p.SequenceNr == 1L && p.Event.Equals(1));
+            probe.ExpectNext<EventEnvelope>(p => p.PersistenceId == PId && p.SequenceNr == 2L && p.Event.Equals(2));
+            await probe.CancelAsync();
+
+            var customReadJournal = Sys.ReadJournalFor<SqlReadJournal>("akka.persistence.query.journal.custom");
+            var customSource = customReadJournal.AllEvents(Offset.NoOffset());
+            var customProbe = customSource.RunWith(this.SinkProbe<EventEnvelope>(), Sys.Materializer());
+            customProbe.Request(2);
+            customProbe.ExpectNext<EventEnvelope>(p => p.PersistenceId == PId && p.SequenceNr == 1L && p.Event.Equals(1));
+            customProbe.ExpectNext<EventEnvelope>(p => p.PersistenceId == PId && p.SequenceNr == 2L && p.Event.Equals(2));
+            await customProbe.CancelAsync();
+
+            // Probe the database directly to make sure that all tables were created properly
+            var tables = await GetTableNamesAsync(_fixture.ConnectionString);
+
+            tables.Should().Contain("journal");
+            tables.Should().Contain("tags");
+            tables.Should().Contain("snapshot");
+            tables.Should().Contain("journal2");
+            tables.Should().Contain("tags2");
+            tables.Should().Contain("snapshot2");
```

Comment:
Testing that all tables were created properly

Reply:
"journal-metadata" table were not created because we're not using compatibility mode

Reply:
LGTM